graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,178 @@
+
+    null_distribution_ : None or ndarray, shape (n_bootstraps, )
         The distribution of T statistics generated under the null.
 
     References
     ----------
-    .. [1] Tang, M., Athreya, A., Sussman, D. L., Lyzinski, V., & Priebe, C. E. (2017). 
+    .. [1] Tang, M., Athreya, A., Sussman, D. L., Lyzinski, V., & Priebe, C. E. (2017).
         "A nonparametric two-sample hypothesis testing problem for random graphs."
         Bernoulli, 23(3), 1599-1630.
+
+    .. [2] Panda, S., Palaniappan, S., Xiong, J., Bridgeford, E., Mehta, R., Shen, C., & Vogelstein, J. (2019).
+        "hyppo: A Comprehensive Multivariate Hypothesis Testing Python Package."
+        arXiv:1907.02088.
     """
 
-    def __init__(self, n_components=None, n_bootstraps=200, bandwidth=None):
+    def __init__(
+        self,
+        test="dcorr",
+        metric="euclidean",
+        n_components=None,
+        n_bootstraps=200,
+        num_workers=1,
+    ):
+
+        if not isinstance(test, str):
+            msg = "test must be a str, not {}".format(type(test))
+            raise TypeError(msg)
+        elif test not in _VALID_TESTS:
+            msg = "Unknown test {}. Valid tests are {}".format(test, _VALID_TESTS)
+            raise ValueError(msg)
+
+        if not isinstance(metric, str) and not callable(metric):
+            msg = "Metric must be str or callable, not {}".format(type(metric))
+            raise TypeError(msg)
+        elif metric not in _VALID_METRICS and not callable(metric):
+            msg = "Unknown metric {}. Valid metrics are {}, or a callable".format(
+                metric, _VALID_METRICS
+            )
+            raise ValueError(msg)
+
         if n_components is not None:
             if not isinstance(n_components, int):
-                msg = "n_components must an int, not {}.".format(type(n_components))
+                msg = "n_components must be an int, not {}.".format(type(n_components))
                 raise TypeError(msg)
 
         if not isinstance(n_bootstraps, int):
-            msg = "n_bootstraps must an int, not {}".format(type(n_bootstraps))
+            msg = "n_bootstraps must be an int, not {}".format(type(n_bootstraps))
             raise TypeError(msg)
         elif n_bootstraps < 1:
             msg = "{} is invalid number of bootstraps, must be greater than 1"
             raise ValueError(msg.format(n_bootstraps))
 
-        if bandwidth is not None and not isinstance(bandwidth, float):
-            msg = "bandwidth must an int, not {}".format(type(bandwidth))
+        if not isinstance(num_workers, int):
+            msg = "num_workers must be an int, not {}".format(type(num_workers))
             raise TypeError(msg)
+        elif num_workers <= 0:
+            msg = "{} is invalid number of workers, must be greater than 0"
+            raise ValueError(msg.format(num_workers))
+        elif num_workers > 1:
+            raise NotImplementedError()  # TODO env error parallelizing
```

Comment:
if not implemented then should not be an option in the docs IMO

Reply:
@bdpedigo I just tried having >1 workers and it worked for me...

Reply:
possibly something archaic, but also may be that my tests weren't extensive enough

Reply:
think we should take out this not implemented?

Reply:
I'd say yes (but add it back if any tests give us error described).

Reply:
tests or notebooks*